tinkerpop3/src/main/java/io/shiftleft/overflowdb/OdbNode.java

```diff
@@ -388,12 +388,16 @@ protected void removeEdge(Direction direction, String label, int blockOffset) {
   }
 
   private Iterator<NodeRef> createAdjacentNodeIterator(Direction direction, String label) {
-    int offsetPos = getPositionInEdgeOffsets(direction, label);
+    return createAdjacentNodeIteratorByOffSet( getPositionInEdgeOffsets(direction, label));
+  }
+
+  // Simplify hoisting of string lookups.
+  // This is on the hot path, hence final.
+public final Iterator<NodeRef> createAdjacentNodeIteratorByOffSet(int offsetPos){
     if (offsetPos != -1) {
       int start = startIndex(offsetPos);
       int length = blockLength(offsetPos);
-      int strideSize = getStrideSize(label);
-
+      int strideSize = layoutInformation().nPropertiesbyOffsetPos[offsetPos] + 1;
```

Comment:
why only use that optimization here, rather than directly in `getStrideSize`, so that other usages benefit from it as well?

Reply:
The idea of this PR is to move all strings (e.g. label) off the hot path. For this I just needed a fast offsetPos->stride map, because the fast API use knows nothing about labels/strings and only knows offsetPos.

Reply:
ok